integrationArtifactTransport Command (#4131)
SAP/jenkins-library: documentation/docs/steps/integrationArtifactTransport.md

# ${docGenStepName}

## ${docGenDescription}

> With this step, you can transport Cloud Integration, capability of SAP Integration Suite content across various landscapes using SAP Content Agent Service.

Author: done

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. for more information on

Reviewer: ```suggestion
SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on
```

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Author: reverted

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Author: there is a separate review happen for documentation part by the Documentation experts, i would suggest not review the .md. .yaml content in detail

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Reviewer 2: Since the documentation is my main way (and probably the main way for other people) to understand what this step is supposed to do, I would like a bit more explanation there if you could add that. When will the documentation experts review this? I can wait to add my review until they've made changes.

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Reviewer 2: After the documentation review, I'm still left with some questions about what exactly this step does and what the documentation is saying. Could you please elaborate on "SAP Cloud Integration provides content transport mechanism."  Is there any other documentation you could link here? Also you mention content providers in the next sentence, but which content providers do you mean?

Since I'm not a Integration Suite expert, my suggestions won't be completely accurate. @AnnikaGonnermann and @axelalbrechtsap maybe you could support here with adding more detail to the documentation from the point of view of Integration Suite? I think it should be written for someone that has no idea what this step should do.

> SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Author: 2 points:
1. "SAP Cloud Integration provides content transport mechanism" used in the already referred public documentation which we also used as is (https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)
2. there is blog linked in the last step which provide step by step instruction how to do integration package transport with CAS, TMS (https://blogs.sap.com/2022/03/25/transport-sap-cloud-integration-ci-cpi-content-with-transport-management-service-tms-and-content-agent-service-cas/)

i feel this is good enough from documentation standpoint.

> Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or can be exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on

Reviewer 2: ```suggestion
SAP Integration Suite provides the ability to transport its content to other services. SAP Content Agent service enables you to assemble the content from various content providers (including SAP Integration Suite) in MTAR format. Later, this content is either available for download or can be exported to a configured transport queue, such as SAP Cloud Transport Management. This step, integrationArtifactTransport, only supports transporting Integration Packages from SAP Integration Suite. For more information on
```

Author: done

> configurations required for SAP Cloud integration, see the documet [Content Assembly for SAP Integration Suite](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)

Reviewer: ```suggestion
configurations required for SAP Cloud integration, see the document [Content Assembly for SAP Integration Suite](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)
```

> configurations required for Cloud Integration, see [Content Assembly for SAP Integration Suite](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)

Reviewer 2: ```suggestion
configurations required for SAP Integration Suite, see [Content Assembly for SAP Integration Suite](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)
```

Author: done

> To consume the integrationArtifactTransport step, proceed as follows:

Author: fixed

> * [Create SAP Content Agent service Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/a4da0c26ced74bbfbc60e7f607dc05ab.html).

Reviewer 2: ```suggestion
* [Create SAP Content Agent Service Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/a4da0c26ced74bbfbc60e7f607dc05ab.html).
```

Author: done

* [Create SAP Cloud Integration Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/c17c4004049d4d9dba373d72ce5610cd.html).

> * [Export using SAP Cloud Transport Management](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/b44463a657fa4be48ea2525b7eb6e7de.html).

Author: fixed

> * Transport Cloud Integration Content with CAS as explained in the blog [Transport Cloud Integration (CI/CPI) Content with Transport Management Service (TMS) and Content Agent service (CAS)](https://blogs.sap.com/2022/03/25/transport-sap-cloud-integration-ci-cpi-content-with-transport-management-service-tms-and-content-agent-service-cas/)

Reviewer 2: ```suggestion
* Transport Cloud Integration Content with SAP Content Agent Service as explained in the blog [TMS – Transport SAP Cloud Integration (CI/CPI) Content with Transport Management Service (TMS) and Content Agent Service (CAS)](https://blogs.sap.com/2022/03/25/transport-sap-cloud-integration-ci-cpi-content-with-transport-management-service-tms-and-content-agent-service-cas/)
```

Author: done

> * integrationArtifactTransport step only supports Integration Package transport.

Author: correct

> * integrationArtifactTransport step only supports Integration Package transport.

Reviewer 2: ```suggestion
```

Author: done

## Prerequisites

## ${docGenParameters}

## ${docGenConfiguration}

## ${docJenkinsPluginDependencies}

## Example

Configuration example for a `Jenkinsfile`:

```groovy
integrationArtifactTransport script: this
```

> Configuration example for a YAML file(for example `.pipeline/config.yaml`):

Reviewer 2: ```suggestion
Configuration example for a YAML file (for example `.pipeline/config.yaml`):
```

Author: done

SAP Cloud Integration provides content transport mechanism. SAP Content Agent service enables you to assemble the content from these content providers in MTAR format. Later, this content is either available for download or exported to the configured transport queue, such as SAP Cloud Transport Management. For more information on
configurations required for SAP Cloud integration, see the document [Content Assembly for SAP Integration Suite](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/8e274fdd41da45a69ff919c0af8c6127.html)

To use the integrationArtifactTransport step, proceed as follows:

* [Create SAP Content Agent Service Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/a4da0c26ced74bbfbc60e7f607dc05ab.html).
* [Create SAP Cloud Integration Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/c17c4004049d4d9dba373d72ce5610cd.html).

> * [Export using SAP Cloud Transport Management Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/b44463a657fa4be48ea2525b7eb6e7de.html).

Reviewer 2: ```suggestion
* [Create SAP Cloud Transport Management Destination](https://help.sap.com/docs/CONTENT_AGENT_SERVICE/ae1a4f2d150d468d9ff56e13f9898e07/b44463a657fa4be48ea2525b7eb6e7de.html).
```
